packages/augur-core/source/libraries/ContractDeployer.ts

```diff
@@ -82,23 +82,25 @@ Deploying to: ${networkConfiguration.networkName}
         const externalAddresses = this.configuration.externalAddresses;
 
         // Legacy REP
-        if (externalAddresses.LegacyReputationToken) {
+        if (this.configuration.isProduction || externalAddresses.LegacyReputationToken) {
+            if (!externalAddresses.LegacyReputationToken) throw new Error('Must provide LegacyReputationToken');
             console.log(`Registering Legacy Rep Contract at ${externalAddresses.LegacyReputationToken}`);
             await this.augur!.registerContract(stringTo32ByteHex('LegacyReputationToken'), externalAddresses.LegacyReputationToken);
         } else {
             await this.uploadLegacyRep();
         }
 
         // REP price oracle
-        if (externalAddresses.RepPriceOracle) {
+        if (this.configuration.isProduction || externalAddresses.RepPriceOracle) {
+            if (!externalAddresses.RepPriceOracle) throw new Error('Must provide RepPriceOracle');
             console.log(`Registering Rep Price Oracle Contract at ${externalAddresses.RepPriceOracle}`);
             await this.augur!.registerContract(stringTo32ByteHex('RepPriceOracle'), externalAddresses.RepPriceOracle);
         } else {
             await this.uploadRepPriceOracle();
         }
 
         // Cash
-        if (externalAddresses.Cash) {
+        if (this.configuration.isProduction || externalAddresses.Cash) {
```

Comment:
It feels like this should be:
```typescript
if (this.configuration.isProduction
    || externalAddresses.Cash
    || externalAddresses.DaiVat
    || externalAddresses.DaiPot
    || externalAddresses.DaiJoin) {
    if (!(externalAddresses.Cash && externalAddresses.DaiVat && externalAddresses.DaiPot && externalAddresses.DaiJoin)) {
        ...
    }
}
```
As it currently stands, if `externalAddresses.DaiVat` is set but `externalAddresses.Cash` is not the code won't make it into this block.  That being said, with the `isProduction` check this is really just a testing thing so not a big deal.

Reply:
Same comment for the other blocks that have multiple requirements.

Reply:
Actually, due to the addition of `isProduction` the check below **MUST** test that `externalAddresses.Cash` is provided.